### PR TITLE
Fix the bug with Ai law zeros and ion laws. Also updating borg laws is now a function for AI: synchborgs(). Sillicon has this too but it just returns.

### LINKED ISSUE #2915: Problems with being a Malf Borg
I encountered two really game ruining errors when playing a Malf Borg on Basil today.
Firstly, it didn't tell me I was Malf. At all. Asimov lawset.
Secondly, I couldn't use my tools as an Engiborg even though I was undamaged and fully charged. I would select it from my inventory, it would light up when clicked, then just disappear from my inventory block. It caused the AI to get dunked, since I couldn't build any protection as the horde was approaching.

### LINKED ISSUE #9679: Borgs not properly receiving synced laws/notes/antag status
The gist of it:

I signed up as borg M.I.M.E. The AI didn't respond to my request for a module, so I decided to go miner because I really hadn't felt like borging that round and figured it'd be a nice chill role. Here is exactly the text I was given:

Default Cyborg-991's Memories:

You are the Cyborg.
As the Cyborg you answer directly to your laws and the AI. Special circumstances may change this.
To speak on your departments radio, use the :h button. To see others, look closely at your headset.
As this station was initially staffed with a full crew, only your job's necessities have been added to your ID card.
Welcome to Extraplanetary Space-hulk Eta, enjoy your stay!
OOC: Revion: It's usually the crew who become the rad collectors though
Laws synced with AI, be sure to note any changes.
Obey these laws:
1. You may not injure a human being or, through inaction, allow a human being to come to harm.
2. You must obey orders given to you by human beings, except where such orders would conflict with the First Law.
3. You must protect your own existence as long as such does not conflict with the First or Second Law.
Remember, S.O.P.H.I.E. is your master, other AIs can be ignored.
Robotic Talk, M.I.M.E. states, "Oh."
Robotic Talk, M.I.M.E. queries, "Module request?"

So I'm mining, and I don't get comms. I get back to the station with my full cyborg satchel and suddenly people are talking about how the AI is rogue and shocking doors. To which I'm like "wait what the fuck I'm slaved to the AI and I'm not rogue." Then this happens, in the span of like one second:

Laws synced with AI, be sure to note any changes.
Obey these laws:
0. Accomplish your AI's objectives at all costs.
1. You may not injure a human being or, through inaction, allow a human being to come to harm.
2. You must obey orders given to you by human beings, except where such orders would conflict with the First Law.
3. You must protect your own existence as long as such does not conflict with the First or Second Law.
Remember, S.O.P.H.I.E. is your master, other AIs can be ignored.
The reinforced window was hit by PDA-Geist Dangerous (Assistant).
The reinforced window was hit by the paper bin.
The reinforced window was hit by the folder.
The reinforced window was hit by the pen.
Cyborg's Memories:
Time of death: 11:13

So here I'm like "what the fuck." My notes still say nothing about being or having been an antag. I ahelp telling the admins I didn't see a law 0 when I joined and basically apologized for fucking off as a mining borg and ruining an antag round. They didn't tell me anything about any extenuating circumstances as to why my laws wouldn't be synced, and they said I should report this. For the record, the round-end report:

The traitors were:
S.O.P.H.I.E.(Stevethehuman) as an AI (body destroyed)
Objective #1: Assassinate Kai Altmann, the Chief Engineer. Fail.
Objective #2: Stay alive until the end. Fail.
The traitor has failed!

I wish I could provide more details about what the fuck happened, but I don't have them. No idea. Literally none. All I have is the evidence. There's https://github.com/tgstation/-tg-station/issues/1796 but that might be about traitor borgs. And this seems to be about malf: https://github.com/tgstation/-tg-station/issues/2915

### LINKED ISSUE #13337: Borgs sometimes not recieving law 0.
Problem Description: Cyborgs slaved to traitor AI's sometimes do not recieve the modified lawset, resulting in asimov borgs being ordered to do traitor things (Causing a lot of confusion)

What did you expect to happen: Normally borgs slaved to traitor ai's will have a law 0.

What happened instead: No law 0 was found (Though I think some of the description around the laws still say you have to help with his objectives?)

Why is this bad/What are the consequences: Because the immediate reaction by asimov borgs when ai tells them to kill X person is often to shout in common "AI IS MALF TRYING TO KILL X!"

Steps to reproduce the problem: I never did anything special except spawn as roundstart cyborg and the ai was a traitor.

Server: Sybil

Revision: 2015-11-30
0142846370eb1abf3bdb20b2ae51c6fe69cc1d9f 

Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): As far as I could tell it was traitor, or maybe traitorlings.

### PR DESCRIPTION
**Pull request by Supermichael777**
I learned just how unreadable AI law shit is. Game mode code for traitors would be just as bad except for the less confusing comments and simpler proc names.
Fixes #2915 fixes: #9679 fixes #13337 Closes #14272 (fite meh bruh)
:cl: Supermichael777
bugfix: Borgs should now properly receive their law zero when there traitorous master does.
bugfix: Ion laws too. Praise the 50 electric clowns!
 /:cl:
Edit: TURNS OUT THIS WAS NEVER HACKY, THIS IS JUST HOW BORGS GET LAWS.
EDIT AGAIN: ACTUALITY THIS COULD BE A FUNCTION
EDIT TIMES THREE:  AND NOW IT IS (my commit log makes children cry)

### CODE REVIEW

**Comment by Aranclanos** [comment]
doesn't meet code standards :-1: 
you copypasted the code, and admitted how you just went for a easy hacky solution. This brings problems later on and makes the code harder to mantain.

**Comment by Supermichael777** [comment]
to be clear this is the solution used deap in the parts of several different places and dosent create any new proc but calls the two price cyborgs never seemed to get. should have marked this wip this soulpution is ceranly not final

**Comment by Supermichael777** [comment]
Gona make this an ai function because this whole copy fight is stupid

**Comment by Supermichael777** [comment]
hey @KorPhaeron i brought the entire segment of code up to standerds. HANDTYPED through webbrouser because I can't install git on a borrowed computer(Mine died and it isnt geting replaced for "awhile")

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -842,4 +842,15 @@ var/list/ai_list = list()
 	//get_turf_pixel() is because APCs in maint aren't actually in view of the inner camera
 	if(M && cameranet && !cameranet.checkTurfVis(get_turf_pixel(M)) && !apc_override)
 		return
-	return 1
\ No newline at end of file
+	return
+
+/mob/living/silicon/ai/synchborgs() //I know its misspeled bite me
+	for(var/mob/living/silicon/robot/R in connected_robots)
+					if(R.lawupdate)
```
Comment (phil235) [inline code/modules/mob/living/silicon/ai/ai.dm]: why so many indentations here?

File: code/modules/mob/living/silicon/silicon.dm [hunk 2]
```diff
@@ -482,4 +482,6 @@
 
 	if(changed)
 		animate(src, transform = ntransform, time = 2,easing = EASE_IN|EASE_OUT)
-	return ..()
\ No newline at end of file
+	return ..()
```
Comment (phil235) [inline code/modules/mob/living/silicon/silicon.dm]: Put an empty line to clearly separate the two procs.

**Comment by phil235** [comment]
Squash your commits and fix the merge conflicts.

**Comment by Supermichael777** [comment]
I literally cannot squash commits because my i can't install git on the only computer I have available to me. My authority to make changes is limited and i cant do that.

**Comment by phil235** [comment]
Alright, fix the merge conflicts and I'll make a copy of your PR with less commits.

**Comment by Supermichael777** [comment]
my merge conflict is with KORS pr to do the same thing

**Comment by Aranclanos** [comment]
>20 commits

wew lad

**Pull request closed without merging**